resources/views/platform/components/game/current-progress/big-status-label.blade.php

```diff
@@ -0,0 +1,49 @@
+@props([
+    'isBeatable' => false,
+    'isBeatenHardcore' => false,
+    'isBeatenSoftcore' => false,
+    'isCompleted' => false,
+    'isMastered' => false,
+    'isEvent' => false,
+])
+
+<?php
+// Determine what color the label should be.
+$colorClassName = "text-text-muted";
+if ($isBeatenSoftcore || $isBeatenHardcore || $isCompleted) {
+    $colorClassName = "text-neutral-300 light:text-neutral-500";
+}
+if ($isMastered) {
+    $colorClassName = "text-yellow-400 light:text-yellow-600";
+}
+
+// Determine what the content of the label should be.
+$statusLabel = "Unfinished";
+if ($isBeatable && $isBeatenSoftcore) {
+    $statusLabel = "Beaten (softcore)";
+}
+if ($isBeatable && $isBeatenHardcore) {
+    $statusLabel = "Beaten";
+}
+if ($isCompleted) {
+    $statusLabel = "Completed";
+}
+if ($isMastered) {
+    $statusLabel = "Mastered";
+}
+if ($isEvent && $statusLabel !== 'Unfinished') {
+    $statusLabel = "Awarded";
+}
+
+// This case can occur on legacy completion/mastery awards
+// where progression achievements were added after the user
+// had already mastered the game. It's an edge case, but we
+// try to gracefully handle it anyway.
+if (($isCompleted || $isMastered) && ($isBeatable && !$isBeatenHardcore && !$isBeatenSoftcore)) {
+    $statusLabel = "Unbeaten";
+}
```

Comment:
I think this is happening incorrectly for my user on game 697.
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/77329fa3-9ae7-479c-9d3c-7dbc5e9b3a60)

Reply:
This is a great catch.

I think what's happening here is you haven't been granted the beaten game award yet for the game, even though you've met the criteria for earning it.

We can detect if we user falls into this circumstance when they visit the game page. I think under that circumstance, you should be granted the award and then would never see this message.

Reply:
In https://github.com/RetroAchievements/RAWeb/pull/1785/commits/df78102fcfa50aa38ece9af939d776056157f552, I've added a fast check on page load for if the user satisfies the beaten game criteria but doesn't have the beaten game award. This will instantly cause `testBeatenGame()` to fire before the page renders.

This isn't perfect. There are still a couple of minute edge cases this won't capture: award revocation and beaten softcore transitioning to beaten hardcore. However, this new logic probably resolves the central theme of this feedback.

Reply:
The first time the page loads, it still shows the Unbeaten text in gold, but it does give the Beaten award, so the next time the page loads it shows Mastered.

Reply:
Ah, I see why this was happening. Taken care of in https://github.com/RetroAchievements/RAWeb/pull/1785/commits/72d211ce105c66751c362b5fdee25e5ee4c06599.